SymPy 1.1 release

asmeurer opened the pull request:
Do not merge this. I will merge it when the release is done.

isuruf committed:
Update symengine tested version and fix deprecation warnings

isuruf committed:
Make USE_SYMENGINE a bool

ixjlyons committed:
Allow custom codegen class to be passed in to autowrap and codegen.

ixjlyons committed:
Cleaned up CCodeGen class customization a bit.

bjodah committed:
F2Py -> f2py

bjodah committed:
Refactor CodeGen to accept printer (see sympy/sympy#12833)

bjodah committed:
Make fortran source format configureable post initialization.

bjodah committed:
Allow custom CodeGen instance to be passed to codegen(...)

asmeurer committed:
Bump release to 1.1.rc1

ixjlyons committed:
Improved custom preprocessor statement interface.

ixjlyons committed:
Added codegen test.

ixjlyons committed:
Revert change in get_code_generator.

asmeurer committed:
Start a Dockerfile for the release

ixjlyons committed:
Don't require newlines in preprocessor_statements.

asmeurer committed:
Fix rever entry point

asmeurer committed:
Split Dockerfile into a base and main file

The main file gets rebuilt every time it is run (from the release.sh script).

Still don't know if this is the best way to do this

asmeurer committed:
Add mailmap_update activity

asmeurer committed:
Import activity

asmeurer committed:
Show full tracebacks from rever.xsh

asmeurer committed:
Make sure release is run from the main sympy directory

asmeurer committed:
Skip the version bump activity in the release script (we already did it manually)

asmeurer committed:
Fix path for mailmap update activity

ixjlyons committed:
Add code_gen arg to autowrap function.

asmeurer committed:
Update mailmap_update.py and authors_update.py to not depend on fabric

They used to require Python 2, but now they require 3.6 (because I don't want
to bother doing the subprocess stuff the hard way).

asmeurer committed:
Update AUTHORS and .mailmap

asmeurer committed:
First stab at copying over build docs from fabfile to rever.xsh

asmeurer committed:
Fix undefined name

asmeurer committed:
Fix undefined name

asmeurer committed:
Actually run the source tarball and build docs activities

asmeurer committed:
Need to install xonda for the rever stuff to work

asmeurer committed:
Apparently xonda isn't installable from conda

asmeurer committed:
xonda has to be installed in the root env, so use rever to do all conda env stuff

asmeurer committed:
Need mpmath to run the mailmap script

asmeurer committed:
Make rever exit on error

asmeurer committed:
Use a version of Sphinx that the docs actually build with

ixjlyons committed:
Added autowrap test.

asmeurer committed:
Conda doesn't have a sphinx 1.3.1 package for Python 3.6

asmeurer committed:
Use version of docutils required to build the docs

asmeurer committed:
Try defining version at the top

isuruf committed:
Use symengine=0.3.*

asmeurer committed:
Merge pull request #12843 from moorepants/compile-args

Compile args

asmeurer committed:
Add copy_release_files to rever.xsh

asmeurer committed:
Merge branch '1.1' of github.com:sympy/sympy into 1.1

ixjlyons committed:
Resolved conflicts with 1.1

ixjlyons committed:
Fixed rust header tests.

asmeurer committed:
Merge pull request #12809 from isuruf/symengine

Update symengine tested version and fix deprecation warnings

asmeurer committed:
Add test tarball activities

asmeurer committed:
Merge branch '1.1' of github.com:sympy/sympy into 1.1

asmeurer committed:
Fix syntax error

asmeurer committed:
Use f-strings and globals() instead of manual formatting

asmeurer committed:
Revert "Use f-strings and globals() instead of manual formatting"

This reverts commit bf5d6e9f8803ceb52be4fec722662956f7b85e88.

Apparently xonsh doesn't support f-strings :(
https://github.com/xonsh/xonsh/issues/2033

asmeurer committed:
Apparently ${...}['VERSION'] does work

asmeurer committed:
Get version working again

asmeurer committed:
Fix lazy tarball format object

asmeurer committed:
Remove references to vagrant, add copy_release_files to activities list

asmeurer committed:
Add ls for debugging

asmeurer committed:
Remove the . before the rc1 in the version

Apparently setuptools removes it, which is causing problems

asmeurer committed:
Print easier to see message in test_tarball

asmeurer committed:
Copy compare tar against git action from the fabfile

asmeurer committed:
Replace an instance of run with $

asmeurer committed:
Run whitelist check before the tarball check

asmeurer committed:
Remove quotes

moorepants commented on .mailmap:
```diff
@@ -284,3 +284,9 @@ Jainul Vaghasia <jainulvaghasia@gmail.com> Jainul Vaghasia <jainulvaghasia@D-10-
 Pranjal Tale <pranjaltale16@gmail.com> = <pranjaltale16@gmail.com>
 jtoka <jason.tokayer@gmail.com> jtoka <jason.tokayer@capitalone.com>
+Theodore Han <theodorehan@hotmail.com> LAPTOP-4VOQQ6GI\han <theodorehan373@gmail.com>
+Theodore Han <theodorehan@hotmail.com> Theodore Han <tqhan317@gmail.com>
```
A repeat.

moorepants replied:
Oh, I forgot the purpose of this. Repeat is fine.

asmeurer replied:
This is .mailmap. Every unique name/email in the git history has to have a line here.

asmeurer committed:
Fix show_files

asmeurer committed:
Copy full_path_split from the fabfile

asmeurer committed:
Add missing os import

asmeurer committed:
Add missing import

asmeurer committed:
Update git whitelist

asmeurer committed:
Better output formatting for whitelist checking

asmeurer committed:
Update tarball whitelist

asmeurer committed:
I think I need a strip() here

asmeurer committed:
Port print_authors from fabfile

asmeurer committed:
Move build_docs back up

It sucks because it's slow, but it has to go before copy_release_files.

asmeurer committed:
Fix syntax error

asmeurer committed:
Properly cd back to the main sympy directory

asmeurer committed:
Add get_short_sympy_version

asmeurer committed:
Add missing get_sympy_version function

asmeurer committed:
Fix sympy version extraction

asmeurer committed:
Remove bold flag, which no longer works

asmeurer committed:
Forgot to import unicodedata

asmeurer committed:
Add md5 activity

asmeurer committed:
Remove unneeded quotes from authors format

asmeurer committed:
Try to fix the md5 activity

asmeurer committed:
Fix get_authors

ixjlyons committed:
Fix autowrap test for Python 2.

asmeurer committed:
Fix cruft from bad merge in fabfile

asmeurer committed:
Port GitHub release from fabfile

asmeurer committed:
Add some dependencies to the activities

asmeurer committed:
Add test SymPy activity

asmeurer committed:
Fix xonsh syntax error

Apparently it doesn't support advanced star unpacking.

asmeurer committed:
Get the version from rever, not SymPy

asmeurer committed:
Move checkout and pull to rever; use just one Dockerfile

asmeurer committed:
Install requests-oauthlib in the Dockerfile

asmeurer committed:
Revert "Move checkout and pull to rever; use just one Dockerfile"

This reverts commit 2a5f2f5a3c89fd6898a418b768f379341cf0a715.

I forgot there's a reason we do this, because we have to checkout and pull
*before* we run rever at all (to get the latest rever.xsh).

asmeurer committed:
Add back OrderedDict import

I don't know how it got removed

asmeurer committed:
Fix typo

ixjlyons committed:
Move preprocessor_statements to instance.

asmeurer committed:
Run the SymPy tests in an environment with all the optional dependencies

asmeurer committed:
Fix rever.xsh syntax

asmeurer committed:
Install some packages that should hopefully fix the conda gcc issue

ixjlyons committed:
Added to AUTHORS.

asmeurer committed:
Merge pull request #12833 from ixjlyons/custom_codegen_class

Allow custom codegen class to be passed in to autowrap and codegen

asmeurer committed:
Skip deprecated modules in doctest

asmeurer committed:
Merge branch '1.1' of github.com:sympy/sympy into 1.1

asmeurer committed:
Fix matplotlib tests in the Dockerfile

asmeurer committed:
Fix doctest failures

asmeurer committed:
Revert "Merge pull request #12711 from siefkenj/commonmatrix7"

This reverts commit 8bb2931b1b60845215a0263dcc1fd745238b89b5, reversing
changes made to 019e47b94dfe0f822913dcc6e193b26ac59fd1f0.

See discussion at https://github.com/sympy/sympy/issues/12389.

asmeurer committed:
Replace an instance of 1.0.1 with 1.1

ixjlyons committed:
Substitute module counter value in expected output.

asmeurer committed:
Revert jordan_cell/jordan_block deprecation

asmeurer committed:
Remove deprecations in matrix module that don't follow the deprecation policy

asmeurer committed:
Add deprecation removal issue for unitsystems deprecation

asmeurer committed:
Add deprecation issue for tensor module

asmeurer committed:
Add deprecation issue for vector module

asmeurer committed:
Merge branch '1.1' into matrix-deprecate-revert

asmeurer committed:
Add deprecation issue for DiracDelta.simplify deprecation

asmeurer committed:
Add deprecation issue for Line.equal

asmeurer committed:
SymPyDeprecationWarning.warn just passes the instance

The way it was done it tries to instantiate another instance of the class. But
message can just be an instance of Warning, so do this instead.

asmeurer committed:
Make the issue argument of SymPyDeprecationWarning required

asmeurer committed:
Add deprecated_since_version to Float(prec) deprecation

Also remove value argument, which doesn't say anything useful.

asmeurer committed:
Make feature and deprecated_since_version required arguments of SymPyDeprecationWarning

Also fix the doctests.

asmeurer committed:
Instead of disabling printing, only run the test with matplotlib

Disabling printing makes it harder to debug issues with the test.

asmeurer committed:
Use np.mean instead of np.average

This gets rid of a FutureWarning.

Fixes #12861.

asmeurer committed:
Properly fix matplotlib tests in the Dockerfile

asmeurer committed:
Fix plotting test failure in Python 3.6

The error message for comparing complex numbers changed, so we needed to
update the code that checked for it.

asmeurer committed:
Merge pull request #12855 from ixjlyons/fix_cython_autowrap_test

Substitute module counter value in expected output.

asmeurer committed:
Show which warnings caused the test to fail

asmeurer committed:
Install older tensorflow in Docker

So the tests pass. See https://github.com/sympy/sympy/issues/12863.

asmeurer committed:
Only use one Dockerfile

Instead of force-rebuilding the main Dockerfile, do the checkout and pull from
a script.

asmeurer committed:
We need to checkout this branch in the Dockerfile for now, since it isn't in master

asmeurer committed:
Add some info for release.sh

asmeurer committed:
Better debugging output on plotting test

asmeurer committed:
Don't try to doctest blacklisted unitsystems module

asmeurer committed:
Merge pull request #12854 from asmeurer/matrix-deprecate-revert

Revert "Merge pull request #12711 from siefkenj/commonmatrix7"

asmeurer committed:
Verbose output on plotting test failure

asmeurer committed:
Hopefully fix the plotting tests

For some reason the same warning is raised multiple times in the release
Docker container.

asmeurer committed:
Merge branch '1.1' of github.com:sympy/sympy into 1.1

asmeurer committed:
Fix wrong variable

szymag committed:
Update documentation.

Remove information about correlation between Del()
and CoordSys3D.

asmeurer committed:
Update git whitelist

asmeurer committed:
Fix error in rever file

asmeurer committed:
Copy prerelease files to the correct directory

asmeurer committed:
Add some changes to make GitHub_release work locally

asmeurer committed:
Fix syntax error in rever.xsh

asmeurer committed:
Some updates to the release README

asmeurer committed:
Fix _print/print_ mixup

asmeurer committed:
Add missing import

asmeurer committed:
Fix undefined name

asmeurer committed:
Add missing import

asmeurer committed:
Fix location of release directory

asmeurer committed:
GitHub activity needs to depend on _version

szymag committed:
Add issue argument

szymag committed:
Merge branch '1.1' into update_docs

bjodah committed:
Fix use of % operator on string with multiple arguments

szymag committed:
Add issue to operators.py

bjodah committed:
Revert change of IndexError

asmeurer committed:
Merge pull request #12871 from szymag/update_docs

Update documentation

smichr commented:
I would really like to see #12881 and #12740 make it into this. They both contain bug fixes. I need #12740 to make piecewise integration work -- there are about a dozen outstanding issues related to that -- and #12881 is a really easy fix to some issues related to Poly free symbols and solving biquadratic system of equations. #12740 is running tests now after a merge.

asmeurer committed:
Merge pull request #12883 from bjodah/fix-matrix-typeerror

Fix use of % operator on string with multiple arguments

asmeurer commented:
I'm sorry, but I'm going to make a call that it's too late for this. I had been planning on making the final release either today or tomorrow, and those changes are rather much to merge post-release candidate. Neither PR seems release critical (correct me if I am wrong). Actually, for the one PR (the piecewise one), I would prefer to see it sit in master for a little while before releasing it, as it seems like a significant change that could use some stabilizing.

asmeurer committed:
Change the version in release.py to 1.1 (final)

smichr commented:
> PR seems release critical (correct me if I am wrong).

It all depends what you think is critical. #12881 is a +47/-21 line change that corrects the ltrim function so it doesn't trim non-dummies (which it does now), corrects `solve_biquadratic` so it actually gives a solution without errors for cases involving infinite solutions, and gives the correct free symbols for a Poly -- it no longer reports dummy symbols that don't appear in the expression. I think those changes are very solid. I'm not going to cry if they don't go in...somebody else might, however, when they get bit by those issues.

I agree that the other PR can sit in master for a while.

asmeurer committed:
Add update_docs from the fabfile

asmeurer committed:
Fix wrong function name

asmeurer committed:
Fix configparser for Python 3

asmeurer committed:
get_sympy_version() should just be version

asmeurer committed:
Fix cd stuff in _update_docs

asmeurer committed:
Fix some more current directory stuff

asmeurer committed:
Proper location for docs tarball

asmeurer committed:
Pull in the docs directory

asmeurer committed:
Add update_sympy_org from fabfile

asmeurer committed:
Fix wrong variable name

asmeurer committed:
Correct release path

asmeurer committed:
Don't make version a local variable

asmeurer committed:
Use absolute path

asmeurer committed:
Update release README

asmeurer committed:
Remove the Vagrantfile

asmeurer committed:
Merge branch 'master' into 1.1

asmeurer commented:
This is ready for merging into master.

moorepants commented:
Nice work on this! Congrats.
